src/main/groovy/edu/oregonstate/mist/locations/db/LibraryDAO.groovy

```diff
@@ -77,31 +86,55 @@ class LibraryDAO {
     }
 
     private Map<String, LibraryHours> getLibraryData(DateTime startDate, int numDays) {
-        def response
         String parameters = getDatesParameter(startDate, numDays)
-        Map<String, LibraryHours> data = new HashMap<>()
 
+        def entityString
         try {
-            HttpPost post = new HttpPost(libraryUrl)
-            post.setHeader("Content-Type", "application/json")
-            post.setHeader("Accept", "application/vnd.kiosks.v1")
+            entityString = doPostRequest(libraryUrl, parameters)
 
-            post.setEntity(new StringEntity(parameters))
+            def data = mapLibraryHours(entityString)
 
-            response = httpClient.execute(post)
-            HttpEntity entity = response.getEntity()
-            def entityString = EntityUtils.toString(entity)
+            if (data.isEmpty()) {
+                throw new DAOException("found no library hours")
+            }
+
+            cache.writeDataToCache(LIBRARY_PATH, entityString)
 
-            data = (Map<String,LibraryHours>)this.mapper.readValue(
-                    entityString,
-                    new TypeReference<HashMap<String, LibraryHours>>() {}
-            )
+            return data
+        } catch (Exception e) {
+            LOGGER.error("Error getting library json data", e)
+            entityString = cache.getCachedData(LIBRARY_PATH)
+            return mapLibraryHours(entityString)
+        }
+    }
 
-            EntityUtils.consume(entity)
+    private Map<String,LibraryHours> mapLibraryHours(String entityString) {
+        (Map<String,LibraryHours>)this.mapper.readValue(entityString,
+                new TypeReference<HashMap<String, LibraryHours>>() {})
+    }
+
+    private String doPostRequest(String url, String body) {
+        HttpPost post = new HttpPost(url)
+        post.setHeader("Content-Type", "application/json")
+        post.setHeader("Accept", "application/vnd.kiosks.v1")
+        post.setEntity(new StringEntity(body))
+
+        CloseableHttpResponse response = null
+        try {
+            response = httpClient.execute(post)
+            int code = response.getStatusLine().getStatusCode()
+            if (code != 200) {
```

Comment:
Are we checking for a `200` in our other classes that get data via http? I think this is a good think to check for. Maybe we're already checking and I just missed it.

Reply:
Yes, I think every other http fetch ends up going through Cache.getData, which checks the response code: https://github.com/osu-mist/locations-api/blob/c1603582e746efb5b0b1f7b53f8a7639f95ea877/src/main/groovy/edu/oregonstate/mist/locations/Cache.groovy#L121.